File: electro-server/src/main/java/com/electro/entity/product/Supplier.java
package com.electro.entity.product;

import com.electro.entity.BaseEntity;
import com.electro.entity.address.Address;
import com.electro.entity.inventory.PurchaseOrder;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.JoinColumn;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.Table;
import java.util.ArrayList;
import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Accessors(chain = true)
@Entity
@Table(name = "supplier")
public class Supplier extends BaseEntity {
    @Column(name = "display_name", nullable = false)
    private String displayName;

    @Column(name = "code", nullable = false, unique = true)
    private String code;

    @Column(name = "contact_fullname")
    private String contactFullname;

    @Column(name = "contact_email")
    private String contactEmail;

    @Column(name = "contact_phone")
    private String contactPhone;

    @Column(name = "company_name")
    private String companyName;

    @Column(name = "tax_code")
    private String taxCode;

    @Column(name = "email")
    private String email;

    @Column(name = "phone")
    private String phone;

    @Column(name = "fax")
    private String fax;

    @Column(name = "website")
    private String website;

    @OneToOne(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @JoinColumn(name = "address_id", referencedColumnName = "id", unique = true)
    private Address address;

    @Column(name = "description")
    private String description;

    @Column(name = "note")
    private String note;

    @Column(name = "status", nullable = false, columnDefinition = "TINYINT")
    private Integer status;

    @OneToMany(mappedBy = "supplier", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference
    private List<Product> products = new ArrayList<>();

    @OneToMany(mappedBy = "supplier", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference
    private List<PurchaseOrder> purchaseOrders = new ArrayList<>();
}

File: electro-client/src/models/Supplier.ts
import BaseResponse from 'models/BaseResponse';
import { AddressRequest, AddressResponse } from 'models/Address';

export interface SupplierResponse extends BaseResponse {
  displayName: string;
  code: string;
  contactFullname: string | null;
  contactEmail: string | null;
  contactPhone: string | null;
  companyName: string | null;
  taxCode: string | null;
  email: string | null;
  phone: string | null;
  fax: string | null;
  website: string | null;
  address: AddressResponse | null;
  description: string | null;
  note: string | null;
  status: number;
}

export interface SupplierRequest {
  displayName: string;
  code: string;
  contactFullname: string | null;
  contactEmail: string | null;
  contactPhone: string | null;
  companyName: string | null;
  taxCode: string | null;
  email: string | null;
  phone: string | null;
  fax: string | null;
  website: string | null;
  address: AddressRequest | null;
  description: string | null;
  note: string | null;
  status: number;
}
